packages/big-design/src/components/Worksheet/Modal/Modal.tsx

```diff
@@ -12,12 +12,19 @@ interface WorksheetModalProps<Item> {
 const InternalWorksheetModal = <T extends WorksheetItem>({ column }: WorksheetModalProps<T>) => {
   const { config, hash } = column;
   const { header, render, saveActionText = 'Save', cancelActionText = 'Cancel' } = config;
+  const { store, useStore } = useWorksheetStore();
 
-  const isModalOpen = useStore(useMemo(() => (state) => state.openedModal === hash, [hash]));
-  const selectedCell = useStore(useMemo(() => (state) => state.selectedCells[0], []));
+  const isModalOpen: boolean = useStore(
+    store,
+    useMemo(() => (state) => state.openedModal === hash, [hash]),
+  );
+  const selectedCell = useStore(
+    store,
+    useMemo(() => (state) => state.selectedCells[0], []),
```

Comment:
Noticed we have some `memos` in our selectors, not sure if we need all of them , but didn't want to remove it for now just in case. Probably we can dedicate more time checking this?